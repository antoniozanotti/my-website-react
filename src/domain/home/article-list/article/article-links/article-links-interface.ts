import { Link } from "@/types/link";

export interface ArticleLinksInterface {
  links?: Link[];
}
