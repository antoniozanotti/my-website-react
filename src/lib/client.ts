import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";

const TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
const SPACE = process.env.CONTENTFUL_SPACE_ID;
const ENVIRONMENT = process.env.CONTENTFUL_ENVIRONMENT_ID;
const URL = `https://graphql.contentful.com/content/v1/spaces/${SPACE}/environments/${ENVIRONMENT}`;

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: URL,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
      },
    }),
  });
});
