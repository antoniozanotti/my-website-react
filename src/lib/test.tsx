import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
