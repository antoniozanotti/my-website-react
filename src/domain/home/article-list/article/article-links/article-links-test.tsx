import { test, expect } from "@playwright/experimental-ct-react";
import { ArticleLinks } from "./article-links";

test.describe("ArticleLinks", () => {
  test("should not render component", async ({ mount }) => {
    const component = await mount(<ArticleLinks />);
    await expect(component.locator(":scope:is(ul)")).toHaveCount(0);
  });

  test("should render ul, li, a", async ({ mount, context }) => {
    const component = await mount(
      <ArticleLinks
        links={[
          { link: "https://www.google.com/", label: "Google" },
          { link: "https://www.linkedin.com/", label: "LinkedIn" },
        ]}
      />
    );
    await expect(component.locator(":scope:is(ul)")).toBeVisible();
    await expect(component.getByRole("listitem")).toHaveCount(2);
    await expect(component.getByRole("link")).toHaveCount(2);
    const pagePromise = context.waitForEvent('page');
    component.getByRole("link").first().click();
    const newPage = await pagePromise;
    await expect(newPage).toHaveURL("https://www.google.com/");
  });
});
