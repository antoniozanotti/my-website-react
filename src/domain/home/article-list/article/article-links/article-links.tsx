import { ArticleLinksInterface } from "./article-links-interface";
import { List } from "@/domain/ui/list/list";
import { styles } from "@/consts/styles";

export function ArticleLinks({ links }: ArticleLinksInterface) {
  return (
    links && (
      <List className="pl-4">
        {links.map((link, index) => (
          <li key={index} className="h-12">
            <a
              href={link.link}
              title={link.label}
              target="_blank"
              className={styles.aHover}
            >
              {link.label}
            </a>
          </li>
        ))}
      </List>
    )
  );
}
