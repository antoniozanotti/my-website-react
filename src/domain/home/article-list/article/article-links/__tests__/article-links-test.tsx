import { screen, render, firstComponent } from "@/lib/test";
import { ArticleLinks } from "../article-links";

describe("ArticleLinks", () => {
  it("should not render component", async () => {
    const { container } = render(<ArticleLinks />);
    const component = firstComponent(screen);
    expect(component).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("should render ul, li, a", async () => {
    const { container } = render(
      <ArticleLinks
        links={[
          { link: "https://www.google.com/", label: "Google" },
          { link: "https://www.linkedin.com/", label: "LinkedIn" },
        ]}
      />
    );
    
    const ul = screen.getByRole("list");
    const lis = screen.getAllByRole("listitem");
    const links = screen.getAllByRole("link");
    
    expect(ul).toBeVisible();
    expect(lis).toHaveLength(2);
    expect(links).toHaveLength(2);

    expect(links[0]).toHaveAttribute("href", "https://www.google.com/");
    expect(links[0]).toHaveAttribute("target", "_blank");
    expect(container).toMatchSnapshot();
  });
});
