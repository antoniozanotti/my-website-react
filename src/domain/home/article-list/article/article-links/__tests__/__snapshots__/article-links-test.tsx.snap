// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleLinks should not render component 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      />
    </div>
  </div>
</div>
`;

exports[`ArticleLinks should render ul, li, a 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <ul
          class="list-disc list-inside pl-4"
        >
          <li
            class="h-12"
          >
            <a
              class="underline text-accent-1 dark:text-dark-accent-1 hover:text-secondary hover:dark:text-dark-secondary"
              href="https://www.google.com/"
              target="_blank"
              title="Google"
            >
              Google
            </a>
          </li>
          <li
            class="h-12"
          >
            <a
              class="underline text-accent-1 dark:text-dark-accent-1 hover:text-secondary hover:dark:text-dark-secondary"
              href="https://www.linkedin.com/"
              target="_blank"
              title="LinkedIn"
            >
              LinkedIn
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
